src/gui/Application.cpp

```diff
@@ -324,25 +327,30 @@ void Application::socketReadyRead()
         return;
     }
 
-    QStringList fileNames;
-    quint32 id;
-    in >> id;
+    SocketCmd id;
+    // manual reinterpret_cast not needed for Qt 5.14+
+    in >> reinterpret_cast<typename std::underlying_type<SocketCmd>::type&>(id);
 
-    // TODO: move constants to enum
     switch (id) {
-    case 1:
+    case SocketCmd::OpenFiles: {
+        QStringList fileNames;
         in >> fileNames;
         for (const QString& fileName : asConst(fileNames)) {
             const QFileInfo fInfo(fileName);
             if (fInfo.isFile() && fInfo.suffix().toLower() == "kdbx") {
                 emit openFile(fileName);
             }
         }
-
         break;
-    case 2:
+    }
+    case SocketCmd::LockAll:
         getMainWindow()->lockAllDatabases();
         break;
+    case SocketCmd::Unlock:
+        QString filename, password, keyfile;
+        in >> filename >> password >> keyfile;
+        emit openFile(filename, password, keyfile);
```

Comment:
This actually let's us fix another issue, passing yubikey information. Can you incorporate that into this change as well?